config/v1/types_oauth.go

```diff
@@ -1,37 +1,420 @@
+	// Setting it to nil means the feature is completely disabled (default)
+	// The default setting can be overriden on OAuthClient basis.
+	// The value represents the maximum amount of time that can occur between
+	// consecutive uses of the token. Tokens become invalid if they are not
+	// used within this temporal window. The user will need to acquire a new
+	// token to regain access once a token times out.
+	// Valid values are:
+	// - 0: Tokens never time out
+	// - X: Tokens time out if there is no activity for X seconds
+	// The current minimum allowed value for X is 300 (5 minutes)
+	AccessTokenInactivityTimeoutSeconds *int32 `json:"accessTokenInactivityTimeoutSeconds,omitempty"`
+}
+
+// IdentityProvider provides identities for users authenticating using credentials
+type IdentityProvider struct {
+	// Name is used to qualify the identities returned by this provider
+	Name string `json:"name"`
+	// UseAsChallenger indicates whether to issue WWW-Authenticate challenges for this provider
+	UseAsChallenger bool `json:"challenge"`
+	// UseAsLogin indicates whether to use this identity provider for unauthenticated browsers to login against
+	UseAsLogin bool `json:"login"`
+	// MappingMethod determines how identities from this provider are mapped to users
+	MappingMethod string `json:"mappingMethod"`
+	// Provider contains the information about how to set up a specific identity provider
+	Provider runtime.RawExtension `json:"provider"`
+
+	// Method determines the default strategy to use when an OAuth client requests a grant.
+	// This method will be used only if the specific OAuth client doesn't provide a strategy
+	// of their own. Valid grant handling methods are:
+	//  - auto:   always approves grant requests, useful for trusted clients
+	//  - prompt: prompts the end user for approval of grant requests, useful for third-party clients
+	//  - deny:   always denies grant requests, useful for black-listed clients
+	// Defaults to "prompt" if not set.
+	Method GrantHandlerType `json:"method"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BasicAuthPasswordIdentityProvider provides identities for users authenticating using HTTP basic auth credentials
+type BasicAuthPasswordIdentityProvider struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// RemoteConnectionInfo contains information about how to connect to the external basic auth server
+	RemoteConnectionInfo `json:",inline"`
+}
+
+// RemoteConnectionInfo holds information necessary for establishing a remote connection
+type RemoteConnectionInfo struct {
+	// URL is the remote URL to connect to
+	URL string `json:"url"`
+	// CA is the CA for verifying TLS connections
+	CA string `json:"ca"`
+	// CertInfo is the TLS client cert information to present
+	// this is anonymous so that we can inline it for serialization
+	CertInfo `json:",inline"`
+}
+
+// CertInfo holds file locations for cert and key
+type CertInfo struct {
```

Comment:
this is an API type, so it can't be a file reference.  It need to be a secret reference.  I can see a strong case for requiring `tls.crt` and `tls.key` to match the kube secret type.

Reply:
moved to secret ref with a note that it should be SecretTypeTLS